requirements-test.txt

```diff
@@ -185,7 +188,9 @@ jsonpatch==1.33
     #   cfn-lint
     #   localstack-core (pyproject.toml)
 jsonpath-ng==1.5.3
-    # via localstack-core (pyproject.toml)
+    # via
+    #   localstack-core (pyproject.toml)
+    #   localstack-snapshot
```

Comment:
Interesting observation here. The pin stays the same because of the transitive dependency/constraint from localstack-snapshot 0.1.0, but for non-test targets the dependency is actually updated.

In our current test setup this will mean the localstack we're running our tests against will have a different version of `jsonpath-ng` than the one that we ship :thinking: 

I'm not sure this is a situation we want. This would be resolved by testing against the actual built container without mounting/installing any test dependencies in there, but for now it might lead to some fairly confusing situations.

This also prompts the question if we should maybe add a way to highlight/restrict these conflicting cases somewhere (e.g. in the periodic dependency updates), since this can continue occurring for other dependencies as well. cc @alexrashed @silv-io

Reply:
Wow, great catch! That is a very good point, and great input for an upcoming iteration on the dependency pinning automation (and maybe the pre-commit hook)!
For this particular case, we should then move forward, merge the upgrade in localstack-snapshot, create a release, and fix the issue hat hand (different versions for testing and publishing).